eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -37,9 +37,9 @@
           }
       },
       "TestMode": {
-          "netcoreapp21_Coverage_Record": {
+          "netcoreapp21_Coverage_or_Record": {
               "SupportsRecording": true,
```

Comment:
Does this matrix entry playing double duty case any extra side effects? As in when we are in recording mode does that get us in any weird state when we are also trying to collect coverage?

Reply:
Good question. @pakrym any knowledge here?

Reply:
To be honest the recording part might be very broken by now considering it's not tested or exercised often. Might be easier to remove it until better days...

Reply:
Seeing as we'll probably need to make things work when we revisit record, I'm in favor of keeping this here so we don't have to re-implement the record variable switching, which is somewhat finicky from a config standpoint.

I think any weird side effects would likely manifest as coverage being lower when record is true, but anyone running record as a manual one-off isn't going to care about coverage.